### PR TITLE
feat: enable configuration of some directory modes for dfdaemon

### PR DESCRIPTION
**Pull request by embroede**
Add the ability to set permissions on dataDir, cacheDir, and workHome 
directories. By default these will use the the previous mode (0755),
but are now configurable if desired by the user.

## Description

Add the ability to set permissions on dataDir, cacheDir, and workHome 
directories. By default these will use the the previous mode (0755),
but are now configurable if desired by the user.

## Related Issue
https://github.com/dragonflyoss/Dragonfly2/issues/2317
## Motivation and Context

Some users are likely to want to be able to lock down permissions of dataDir, cacheDir etc. for clients running dfdaemon, specifically in multi-user environments.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

**Comment by gaius-qi** [comment]
@embroede Please fix CI,

File: client/config/peerhost.go [hunk 1]
```diff
@@ -52,12 +52,15 @@ type DaemonOption struct {
...
 	Metrics    string        `mapstructure:"metrics" yaml:"metrics"`
 
-	WorkHome    string `mapstructure:"workHome" yaml:"workHome"`
-	CacheDir    string `mapstructure:"cacheDir" yaml:"cacheDir"`
-	LogDir      string `mapstructure:"logDir" yaml:"logDir"`
-	PluginDir   string `mapstructure:"pluginDir" yaml:"pluginDir"`
-	DataDir     string `mapstructure:"dataDir" yaml:"dataDir"`
-	KeepStorage bool   `mapstructure:"keepStorage" yaml:"keepStorage"`
+	WorkHome     string `mapstructure:"workHome" yaml:"workHome"`
+	WorkHomeMode int    `mapstructure:"workHomeMode" yaml:"workHomeMode"`
```
Comment (jim3ma) [inline client/config/peerhost.go:56]: Please use `uint32`, same with os.FileMode

File: client/config/peerhost.go [hunk 2]
```diff
@@ -213,6 +216,18 @@ func (p *DaemonOption) Validate() error {
 		}
 	}
 
+	if os.FileMode(p.WorkHomeMode) == 0 {
+		return errors.New("invalid value for WorkHomeMode")
```
Comment (jim3ma) [inline client/config/peerhost.go:220]: If not set, use default mode for compatibility

Comment (embroede) [reply client/config/peerhost.go]: Ok that sounds good. Since that check is already being done where these variables are used, I will remove these checks

**Comment by embroede** [comment]
I'll add tests to bring the code coverage back up

File: client/config/peerhost.go [hunk 3]
```diff
@@ -52,12 +52,15 @@ type DaemonOption struct {
...
-	CacheDir    string `mapstructure:"cacheDir" yaml:"cacheDir"`
-	LogDir      string `mapstructure:"logDir" yaml:"logDir"`
-	PluginDir   string `mapstructure:"pluginDir" yaml:"pluginDir"`
-	DataDir     string `mapstructure:"dataDir" yaml:"dataDir"`
-	KeepStorage bool   `mapstructure:"keepStorage" yaml:"keepStorage"`
+	WorkHome     string `mapstructure:"workHome" yaml:"workHome"`
+	WorkHomeMode uint32 `mapstructure:"workHomeMode" yaml:"workHomeMode"`
+	CacheDir     string `mapstructure:"cacheDir" yaml:"cacheDir"`
+	CacheDirMode uint32 `mapstructure:"cacheDirMode" yaml:"cacheDirMode"`
+	LogDir       string `mapstructure:"logDir" yaml:"logDir"`
```
Comment (gaius-qi) [inline client/config/peerhost.go:59]: Add Log

**Review by jim3ma — APPROVED** [review]
LGTM

**Review by gaius-qi — APPROVED** [review]
LGTM

**Pull request merged**